src/main/java/com/metriql/warehouse/clickhouse/ClickhouseDataSource.kt

```diff
@@ -26,7 +26,15 @@ class ClickhouseDataSource(override val config: ClickhouseWarehouse.ClickhouseCo
     override val dataSourceProperties: Properties by lazy {
         val properties = Properties()
         val port = if (config.port === 9000) 8123 else config.port
-        properties["jdbcUrl"] = "jdbc:clickhouse://${config.host}:$port/${config.database}"
+
+        var queryParams = ""
+        config.driverUriParameters?.let {
+            val list = ArrayList<String>()
+            it.map { (k, v) -> list.add("${k}=${v}") }
+            queryParams = list.joinToString(prefix = "?", separator = "&")
+        }
+        properties["jdbcUrl"] = "jdbc:clickhouse://${config.host}:$port/${config.database}${queryParams}"
+
```

Comment:
Seems like, In order to pass socket_timeout, a query parameter must be used.

Reply:
Yes but the codebase automatically generates the query parameters and adds them automatically. Looks like the issue is different, I will check again and update the PR with my comment.